src/board.cpp

```diff
@@ -0,0 +1,166 @@
+  std::vector<bool> moves(boardSize);
+  std::string result = "";
+  for (int number : playerSet) {
+    moves[number - 1] = true;
+  }
+  for (int i = 0; i < boardSize; i++) {
+    if (moves[i]) {
+      result += std::to_string(i + 1);
+    } else {
+      result += i <= 9 ? "_" : "_ ";
+    }
+    result += "  ";
+  }
+  return result;
+}
+
+void Board::printPlayerMoves(int player) {
+  printHeader("NUMBERS ROYALE");
+  if (player == P1) {
+    printItem("Player 1 moves:");
+  } else {
+    printItem("Player 2 moves:");
+  }
+  printf("| %*s |\n", INNER_WIDTH-INNER_PAD, getPlayerMoves(player).c_str());
+  printBorder();
+}
+
+/**
+ * Note: For debugging purposes
+ * Shows the current state of the board
+ */
+void Board::printBoard(int player = P1) {
+  printHeader("<debug> NUMBERS ROYALE");
+  std::string p2PointsText = "";
+  std::string p1PointsText = "";
+  if (!isCpu) {
+    p2PointsText = "Player 2: " + std::to_string(p2Score) + " points";
+  } else {
+    p2PointsText = "CPU: " + std::to_string(p2Score) + " points";
+  }
+  printItem(p2PointsText.c_str());
+  printf("| %*s |\n", INNER_WIDTH-INNER_PAD, getPlayerMoves(P2).c_str());
+  printItem("");
+  printItem("");
+  printf("| %*s |\n", INNER_WIDTH-INNER_PAD, getPlayerMoves(P1).c_str());
+  p1PointsText = "Player 1: " + std::to_string(p1Score) + " points";
+  printItem(p1PointsText.c_str());
+  printBorder();
+}
+
+void Board::printScore() {
+  int winner = checkWinner();
+  if (winner == P1) {
+    printHeader("PLAYER 1 WINS!");
+  } else if (winner == P2) {
+    if (isCpu) {
+      printHeader("CPU WINS!");
+    } else {
+      printHeader("PLAYER 2 WINS!");
+    }
+  } else if (winner == 0) {
+    printHeader("TIE GAME!");
+  } else {
+    printHeader("CURRENT SCORE", false);
+  }
+  std::string p1ScoreText = "Player 1: " + std::to_string(p1Score) + " points";
+  std::string p2ScoreText = isCpu ? "CPU: " : "Player 2: ";
+  p2ScoreText += std::to_string(p2Score) + " points";
+
+  if (winner == P2) {
+    printItem(p2ScoreText.c_str());
+    printItem(p1ScoreText.c_str());
+  } else {
+    // default order
+    printItem(p1ScoreText.c_str());
+    printItem(p2ScoreText.c_str());
+  }
+  printBorder();
+}
+
+/**
+ * Reset the board to its initial state
+ * @param size the size of the board
+ */
+void Board::reset(int size) {
+  if (size > 0) {
+    boardSize = size;
+  }
+  p1.clear();
+  p2.clear();
+  p1Score = 0;
+  p2Score = 0;
+  for (int i = 1; i <= boardSize; i++) {
+    p1.insert(i);
+    p2.insert(i);
+  }
+}
+
+/**
+ * @return 
+ * - 0: tie game
+ * - 1: player 1 wins
+ * - 2: player 2 wins
+ * - -1: game is still in progress
+ */
+int Board::checkWinner() {
+    int toWin = boardSize / 2;
+    if (p1Score > toWin) {
+        return P1;
+    }
+    if (p2Score > toWin) {
+        return P2;
+    }
+    if (p1.empty() && p2.empty()) {
+        if (p1Score == p2Score) {
+            return 0;
+        }
+        return p1Score > p2Score ? P1 : P2;
+    }
+    return -1;
+}
```

Comment:
newline at eof

Reply:
You didn't fix

Reply:
lol I didn't specify this in the commit, haven't touched board or main yet